### PR TITLE
test: every action in a meta transaction

### PR DESCRIPTION
**Pull request by jakmeier**
*almost every action: CreateAccount is not included because it doesn't 
work with meta transactions, a known limitation. I will add such tests
with the onboarding scenarios.

### CODE REVIEW

File: integration-tests/src/tests/client/features/delegate_action.rs [hunk 1]
```diff
@@ -244,9 +257,119 @@ fn meta_tx_fn_call() {
...
+    let node = RuntimeNode::new(&relayer);
+    let fee_helper = fee_helper(&node);
+
+    let tx_cost = fee_helper.add_key_full_cost();
+    // any public key works as long as it doesn't exists on the receiver, the
+    // relayer public key is just handy
+    let public_key = node.signer().public_key();
+    let actions =
+        vec![Action::AddKey(AddKeyAction { public_key, access_key: AccessKey::full_access() })];
+    check_meta_tx_no_fn_call(&node, actions, tx_cost, 0, sender, relayer, receiver);
```
Comment (mm-near) [inline integration-tests/src/tests/client/features/delegate_action.rs:320]: can you also check that the key was added. ?
(if we have an easy way of doing so)

File: integration-tests/src/tests/client/features/delegate_action.rs [hunk 2]
```diff
@@ -244,9 +257,119 @@ fn meta_tx_fn_call() {
...
+    let relayer = alice_account();
+    // Can only delete keys on own account
+    let receiver = sender.clone();
+    let node = RuntimeNode::new(&relayer);
+    let fee_helper = fee_helper(&node);
+
+    let tx_cost = fee_helper.delete_key_cost();
+    let public_key = PublicKey::from_seed(KeyType::ED25519, &receiver);
+    let actions = vec![Action::DeleteKey(DeleteKeyAction { public_key })];
+    check_meta_tx_no_fn_call(&node, actions, tx_cost, 0, sender, relayer, receiver);
```
Comment (mm-near) [inline integration-tests/src/tests/client/features/delegate_action.rs:335]: check that key was deleted.

File: integration-tests/src/tests/client/features/delegate_action.rs [hunk 3]
```diff
@@ -244,9 +257,119 @@ fn meta_tx_fn_call() {
...
+        check_meta_tx_execution(&node, actions, sender, relayer, receiver);
+
+    assert_eq!(
+        sender_diff,
+        -(balance as i128),
+        "sender should be deleted and thus have zero balance"
+    );
+    assert_eq!(sender_diff, receiver_diff);
+    assert_eq!(relayer_diff, balance as i128 - (gas_cost as i128), "unexpected relayer balance");
+}
```
Comment (mm-near) [inline integration-tests/src/tests/client/features/delegate_action.rs:375]: can you check that account is removed ?

**Review by mm-near — APPROVED** [review]

File: integration-tests/src/tests/client/features/delegate_action.rs (hunk 1 above)
Comment (jakmeier) [reply integration-tests/src/tests/client/features/delegate_action.rs]: good call, yes it's pretty easy

File: integration-tests/src/tests/client/features/delegate_action.rs (hunk 2 above)
Comment (jakmeier) [reply integration-tests/src/tests/client/features/delegate_action.rs]: done

File: integration-tests/src/tests/client/features/delegate_action.rs (hunk 3 above)
Comment (jakmeier) [reply integration-tests/src/tests/client/features/delegate_action.rs:397]: done

**Pull request merged**